### PR TITLE
feat(pharmacy selection): ratings and hours

### PR DESCRIPTION
**Pull request by jknapp25**
Added:
- google ratings
- google hours (closes, opens, open now, permanent/temporary close, lunch window, 24hr)
- better error handling on multiple queries

Tested:

- Locations all across the US (portland, chicago, arizona, new york, new jersey, miami, alabama, south dakota, colorado)
- Slow 3G
- Hitting query limit
- Returning all pharmacies nearby

### CODE REVIEW

File: apps/patient/package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "chakra-react-select": "^4.6.0",
     "graphql-request": "^5.2.0",
+    "moment": "^2.29.4",
```
Comment (captDaylight) [inline apps/patient/package.json:8]: Should we choose another library? From their readme:

```
Moment.js is a legacy project, now in maintenance mode. In most cases, you should choose a different library.
```

File: apps/patient/src/components/PharmacyCard.tsx [hunk 2]
```diff
@@ -26,16 +29,71 @@ interface PharmacyCardProps {
...
+      is24Hr?: boolean
+      opens?: string
+      closes?: string
+    }
   }
   selected: boolean
   onSelect: Function
 }
 
+const Metadata1 = ({ businessStatus, rating, hours }) => {
```
Comment (captDaylight) [inline apps/patient/src/components/PharmacyCard.tsx:45]: Can we give this a more descriptive name?

File: apps/patient/src/components/PharmacyCard.tsx [hunk 3]
```diff
@@ -26,16 +29,71 @@ interface PharmacyCardProps {
...
+      {!open && opens ? (
+        <Text fontSize="sm" color="gray.500">
+          Opens {moment(opens, 'HHmm').format(moment(opens, 'HHmm').minute() > 0 ? 'h:mmA' : 'hA')}
+        </Text>
+      ) : null}
+    </HStack>
+  )
+}
+
+const Metadata2 = ({ distance, address }) => {
```
Comment (captDaylight) [inline apps/patient/src/components/PharmacyCard.tsx:83]: More descriptive name

File: apps/patient/src/components/PharmacyList.tsx [hunk 4]
```diff
@@ -1,9 +1,16 @@
-import { Button, SlideFade, VStack } from '@chakra-ui/react'
+import { Button, SlideFade, Text, VStack } from '@chakra-ui/react'
 
 import { PharmacyCard } from '../components/PharmacyCard'
 import t from '../utils/text.json'
```
Comment (captDaylight) [inline apps/patient/src/components/PharmacyList.tsx:4]: Why are we keeping these bits of copy in this file?

Comment (jknapp25) [reply apps/patient/src/components/PharmacyList.tsx:4]: Just my preferred way of handling copy. Nice to not have long strings in code, if we ever add languages copy needs to be centralized, and helps keep grammar/tone consistent.

File: apps/patient/src/components/PharmacyList.tsx [hunk 5]
```diff
@@ -1,9 +1,16 @@
...
 
-export const PharmacyList = ({ pharmacies, selectedId, handleSelect, handleShowMore }) => {
+export const PharmacyList = ({
+  pharmacies,
+  selectedId,
+  handleSelect,
+  handleShowMore,
+  loadingMore,
+  showingAllPharmacies
+}) => {
```
[the remark this answers is not in the document]
Comment (jknapp25) [reply apps/patient/src/components/PharmacyList.tsx:13]: @captDaylight can i re-use from outside patient?

File: apps/patient/src/components/PharmacyCard.tsx (hunk 2 above)
Comment (jknapp25) [reply apps/patient/src/components/PharmacyCard.tsx:45]: Yeah, I was having trouble coming up with something better than Row1 lol. Any ideas?

Comment (jknapp25) [reply apps/patient/src/components/PharmacyCard.tsx:45]: Ah, I'll do RatingHours

File: apps/patient/src/components/PharmacyCard.tsx (hunk 3 above)
Comment (jknapp25) [reply apps/patient/src/components/PharmacyCard.tsx:83]: Updating to DistanceAddress

File: apps/patient/package.json (hunk 1 above)
Comment (jknapp25) [reply apps/patient/package.json:8]: Sure, I'll explore a bit

File: apps/patient/src/views/Pharmacy.tsx [hunk 6]
```diff
@@ -77,34 +147,77 @@ export const Pharmacy = () => {
...
+        const name = results.pharmaciesByLocation[i].name
+        const address = results.pharmaciesByLocation[i].address
+          ? formatAddress(results.pharmaciesByLocation[i].address)
+          : ''
+
+        const placeRequest = {
+          query: name + ' ' + address,
+          fields: ['place_id']
+        }
+        const { response: place, status: placeStatus }: any = await query(
```
Comment (captDaylight) [inline apps/patient/src/views/Pharmacy.tsx:185]: are there any types for place and  placesStatus?

File: apps/patient/src/components/PharmacyList.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (captDaylight) [reply apps/patient/src/components/PharmacyList.tsx:13]: I don't quite follow

File: apps/patient/src/views/Pharmacy.tsx (hunk 6 above)
Comment (jknapp25) [reply apps/patient/src/views/Pharmacy.tsx]: I'll work on finding some. Wasn't a quick.

File: apps/patient/src/components/PharmacyCard.tsx (hunk 2 above)
Comment (jknapp25) [reply apps/patient/src/components/PharmacyCard.tsx]: udpated

**Review by captDaylight — APPROVED** [review]
👍

**Pull request merged**